sound/soc/sof/intel/hda.c

```diff
@@ -499,6 +499,27 @@ static const struct sof_intel_dsp_desc
 	return chip_info;
 }
 
+irqreturn_t hda_dsp_interrupt_handler(int irq, void *context)
+{
+	if ((hda_dsp_ipc_irq_handler(irq, context) == IRQ_WAKE_THREAD) ||
+	    (hda_dsp_stream_interrupt(irq, context) == IRQ_WAKE_THREAD))
```

Comment:
I would change the return type of those functions to a boolean, e.g.

````c
if (hda_dsp_check_ipc_irq(irq, context) || 
    hda_dsp_check_stream_irq(irq, context))
        return IRQ_WAKE_THREAD;
````

Reply:
@plbossart Done. But I add a new function instead of changing  hda_dsp_ipc_irq_handler() since it is used in somewhere else.